### PR TITLE
Upload images

### PR DESCRIPTION
**Pull request by satya164**

### CODE REVIEW

File: lib/s3-upload.es6 [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+		this._policy.catch(err => {
+			if (typeof this.onerror === "function") {
+				this.onerror(err);
+			}
+		});
+
+		this._request = new XMLHttpRequest();
+	}
+
+	_generateThumb(event) {
```
Comment (aravindet) [inline lib/s3-upload.es6]: pollThumbUrl

File: lib/s3-upload.es6 [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+			let filename;
+
+			switch (this._opts.uploadType) {
+			case "avatar":
+			case "banner":
+				filename = "original." + file.name.split(".").pop();
+				break;
+			case "content":
+				filename = "1/${filename}";
```
Comment (aravindet) [inline lib/s3-upload.es6]: "1/" + file.name

File: lib/s3-upload.es6 [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+			let baseurl = "https://" + policy.bucket + ".s3.amazonaws.com/";
+
+			this._request.addEventListener("load", event => {
+				let path = this._opts.uploadType + "/" + this._opts.userId + "/";
+
+				if (this._opts.uploadType === "content") {
+					path += this._opts.textId + "/1/";
+				}
+
+				this.url = baseurl + encodeURIComponent("uploaded/" + path) + file.name.replace(/\s/g, "+");
```
Comment (aravindet) [inline lib/s3-upload.es6]: this.url = baseUrl + policy.keyPrefix + filename;
this.thumb = baseUrl + policy.keyPrefix.replace(/^uploaded/, "generated") + thumbname;

File: lib/s3-upload.es6 [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+			for (let field of fields) {
+				formData.append(field, policy[field]);
+			}
+
+			let filename;
+
+			switch (this._opts.uploadType) {
+			case "avatar":
+			case "banner":
+				filename = "original." + file.name.split(".").pop();
```
Comment (aravindet) [inline lib/s3-upload.es6]: thumbName = '128x128.jpg';

File: lib/s3-upload.es6 [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+			}
+
+			formData.append("key", key);
+			formData.append("success_action_status", "201");
+			formData.append("file", file);
+
+			let baseurl = "https://" + policy.bucket + ".s3.amazonaws.com/";
+
+			this._request.addEventListener("load", event => {
+				this.url = baseurl + encodeURIComponent(key);
```
Comment (satya164) [inline lib/s3-upload.es6]: If I don't encode the path, then filename with spaces don't work.

Foe example <https://scrollback-1.s3.amazonaws.com/uploaded/content/satya164/fc12e3882200fa6565011567fd4f2c5c/1/upload+file.jpeg> doesn't work, while <https://scrollback-1.s3.amazonaws.com/uploaded%2Fcontent%2Fsatya164%2Ffc12e3882200fa6565011567fd4f2c5c%2F1%2Fupload%2Bfile.jpeg> (`"https://scrollback-1.s3.amazonaws.com/" + encodeURIComponent("uploaded/content/satya164/fc12e3882200fa6565011567fd4f2c5c/1/upload+file.jpeg")`) works.

**Pull request merged**